From d66683c0e536664a8846cc0b68f83303560f7bea Mon Sep 17 00:00:00 2001
From: Zhikui Ren <zhikui.ren@intel.com>
Date: Mon, 10 Jul 2023 13:21:53 -0700
Subject: [PATCH] psusensor:only call checkEventLimits for psu

There are other devices and sensors monitored by psusensor. EventLimits
were designed for power supply only. Limit call to sensor type pmbus,
which is for power supply.

Also removed debug prints that floods the journal log.

Tested:
Use debug prints verified that the function is called only when
creating power supply sensors.

Signed-off-by: Zhikui Ren <zhikui.ren@intel.com>
Upstream-Status: Pending
---
 src/PSUSensor.cpp     | 2 --
 src/PSUSensorMain.cpp | 2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/PSUSensor.cpp b/src/PSUSensor.cpp
index 763a926..4b22fda 100644
--- a/src/PSUSensor.cpp
+++ b/src/PSUSensor.cpp
@@ -150,7 +150,6 @@ void PSUSensor::restartRead(void)
     waitTimer.async_wait([weakRef](const boost::system::error_code& ec) {
         if (ec == boost::asio::error::operation_aborted)
         {
-            std::cerr << "Failed to reschedule\n";
             return;
         }
         std::shared_ptr<PSUSensor> self = weakRef.lock();
@@ -188,7 +187,6 @@ void PSUSensor::handleResponse(const boost::system::error_code& err,
     }
     catch (const std::invalid_argument&)
     {
-        std::cerr << "Could not parse  input from " << path << "\n";
         incrementError();
     }
 
diff --git a/src/PSUSensorMain.cpp b/src/PSUSensorMain.cpp
index f6e5691..fb35c58 100644
--- a/src/PSUSensorMain.cpp
+++ b/src/PSUSensorMain.cpp
@@ -868,7 +868,7 @@ static void createSensorsCallback(
                 }
             }

-            if (devType == DevTypes::HWMON)
+            if (sensorType == "pmbus")
             {
                 checkEventLimits(sensorPathStr, limitEventMatch, eventPathList);
             }
-- 
2.25.1

